pkg/asset/ignition/bootstrap/bootstrap.go

```diff
@@ -43,147 +43,98 @@ type bootstrapTemplateData struct {
-		apiServerCertKey:          tlsStock.APIServerCertKey(),
-		openshiftAPIServerCertKey: tlsStock.OpenshiftAPIServerCertKey(),
-		apiServerProxyCertKey:     tlsStock.APIServerProxyCertKey(),
-		adminCertKey:              tlsStock.AdminCertKey(),
-		kubeletCertKey:            tlsStock.KubeletCertKey(),
-		mcsCertKey:                tlsStock.MCSCertKey(),
-		serviceAccountKeyPair:     tlsStock.ServiceAccountKeyPair(),
-		kubeconfig:                kubeconfigStock.KubeconfigAdmin(),
-		kubeconfigKubelet:         kubeconfigStock.KubeconfigKubelet(),
-		manifests:                 manifestStock.Manifests(),
-		tectonic:                  manifestStock.Tectonic(),
-		kubeCoreOperator:          manifestStock.KubeCoreOperator(),
-	}
-}
+var _ asset.WritableAsset = (*Bootstrap)(nil)
 
-// Dependencies returns the assets on which the bootstrap asset depends.
-func (a *bootstrap) Dependencies() []asset.Asset {
+// Dependencies returns the assets on which the Bootstrap asset depends.
+func (a *Bootstrap) Dependencies() []asset.Asset {
 	return []asset.Asset{
-		a.installConfig,
-		a.rootCA,
-		a.etcdCA,
-		a.ingressCertKey,
-		a.kubeCA,
-		a.aggregatorCA,
-		a.serviceServingCA,
-		a.clusterAPIServerCertKey,
-		a.etcdClientCertKey,
-		a.apiServerCertKey,
-		a.openshiftAPIServerCertKey,
-		a.apiServerProxyCertKey,
-		a.adminCertKey,
-		a.kubeletCertKey,
-		a.mcsCertKey,
-		a.serviceAccountKeyPair,
-		a.kubeconfig,
-		a.kubeconfigKubelet,
-		a.manifests,
-		a.tectonic,
-		a.kubeCoreOperator,
+		&installconfig.InstallConfig{},
+		&tls.RootCA{},
+		&tls.EtcdCA{},
+		&tls.IngressCertKey{},
+		&tls.KubeCA{},
+		&tls.AggregatorCA{},
+		&tls.ServiceServingCA{},
+		&tls.ClusterAPIServerCertKey{},
+		&tls.EtcdClientCertKey{},
+		&tls.APIServerCertKey{},
+		&tls.OpenshiftAPIServerCertKey{},
+		&tls.APIServerProxyCertKey{},
+		&tls.AdminCertKey{},
+		&tls.KubeletCertKey{},
+		&tls.MCSCertKey{},
+		&tls.ServiceAccountKeyPair{},
+		&kubeconfig.Admin{},
+		&kubeconfig.Kubelet{},
+		&manifests.Manifests{},
+		&manifests.Tectonic{},
+		&manifests.KubeCoreOperator{},
 	}
 }
 
-// Generate generates the ignition config for the bootstrap asset.
-func (a *bootstrap) Generate(dependencies map[asset.Asset]*asset.State) (*asset.State, error) {
-	installConfig, err := installconfig.GetInstallConfig(a.installConfig, dependencies)
-	if err != nil {
-		return nil, err
-	}
+// Generate generates the ignition config for the Bootstrap asset.
+func (a *Bootstrap) Generate(dependencies asset.Parents) error {
+	installConfig := &installconfig.InstallConfig{}
+	dependencies.Get(installConfig)
 
-	templateData, err := a.getTemplateData(installConfig)
+	templateData, err := a.getTemplateData(installConfig.Config)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	config := igntypes.Config{
+	a.config = &igntypes.Config{
 		Ignition: igntypes.Ignition{
 			Version: igntypes.MaxVersion.String(),
 		},
 	}
 
-	a.addBootstrapFiles(&config, dependencies)
```

Comment:
This seemed better, than the implicit addition to config.

Reply:
I look at it as adding the bootstrap files to the state of the asset. As you state, adding to the ignition config is implicit in that, which I am fine with. However, I am not strongly opposed to changing it back to passing in the ignition config. Although in that case, I would not want the function to be a member of the Bootstrap type and instead be a standalone function.